pkg/controllers/trusted_ca_bundle_controller.go

```diff
@@ -0,0 +1,216 @@
+			klog.Infof("changed config map %s is not a proxy trusted ca, skipping", req)
+			return reconcile.Result{}, nil
+		}
+	}
+
+	systemTrustBundle, err := r.getSystemTrustBundle()
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get system trust bundle: %v", err)
+	}
+	CCMTrustedConfigMap = r.makeCABundleConfigMap(systemTrustBundle)
+
+	if isSpecTrustedCASet(&proxyConfig.Spec) {
+		userCABundle, err := r.getUserCABundle(ctx, proxyConfig.Spec.TrustedCA.Name)
+		if err != nil {
+			klog.Warningf("failed to get user defined trust bundle, system CA will be used: %v", err)
+		} else {
+			mergedTrustBundle, err := r.mergeCABundles(userCABundle, systemTrustBundle)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("can not merge system and user trust bundles: %v", err)
+			}
+			CCMTrustedConfigMap = r.makeCABundleConfigMap(mergedTrustBundle)
+		}
+	}
+
+	if err := r.createOrUpdateConfigMap(ctx, CCMTrustedConfigMap); err != nil {
+		return reconcile.Result{}, fmt.Errorf("can not update target trust bundle configmap: %v", err)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *TrustedCABundleReconciler) getUserCABundle(ctx context.Context, trustedCA string) ([]byte, error) {
+	cfgMap, err := r.getUserCABundleConfigMap(ctx, trustedCA)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate configmap reference for proxy trustedCA '%s': %v",
+			trustedCA, err)
+	}
+
+	_, bundleData, err := r.getCABundleConfigMapData(cfgMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate trust bundle for proxy trustedCA '%s': %v",
+			trustedCA, err)
+	}
+
+	return bundleData, nil
+}
+
+func (r *TrustedCABundleReconciler) getUserCABundleConfigMap(ctx context.Context, trustedCA string) (*corev1.ConfigMap, error) {
+	cfgMap := &corev1.ConfigMap{}
+	ns := OpenshiftConfigNamespace
+	if err := r.Get(ctx, types.NamespacedName{Namespace: ns, Name: trustedCA}, cfgMap); err != nil {
+		return nil, fmt.Errorf("failed to get trustedCA configmap for proxy %s: %v", proxyResourceName, err)
+	}
+
+	return cfgMap, nil
+}
+
+func (r *TrustedCABundleReconciler) getCABundleConfigMapData(cfgMap *corev1.ConfigMap) ([]*x509.Certificate, []byte, error) {
+	certBundle, bundleData, err := util.TrustBundleConfigMap(cfgMap)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return certBundle, bundleData, nil
+}
+
+func (r *TrustedCABundleReconciler) makeCABundleConfigMap(trustBundle []byte) *corev1.ConfigMap {
+
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      trustedCAConfigMapName,
+			Namespace: r.TargetNamespace,
+		},
+		Data: map[string]string{
+			trustedCABundleConfigMapKey: string(trustBundle),
+		},
+	}
+}
+
+func (r *TrustedCABundleReconciler) createOrUpdateConfigMap(ctx context.Context, cm *corev1.ConfigMap) error {
+	// check if target config exists, create if not
+	err := r.Get(ctx, client.ObjectKeyFromObject(cm), &corev1.ConfigMap{})
+
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
Would remove this empty line, pair the declaration of the error with the check :)